core/src/main/java/greencity/config/DatasourceMetadata.java

```diff
@@ -1,29 +1,39 @@
 package greencity.config;
 
-import java.time.ZoneId;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.event.EventListener;
+import org.springframework.core.io.ClassPathResource;
 import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+
+import javax.sql.DataSource;
+import java.time.ZoneId;
 
 /**
  * Configuration class that obtains metadata of the database.
+ *
  * @author Yurii Koval
  */
 @Slf4j
 @Configuration
 public class DatasourceMetadata {
     private final JdbcTemplate jdbcTemplate;
+    private final DataSource dataSource;
 
     /**
      * Constructor.
      *
      * @param jdbcTemplate {@link JdbcTemplate}
      */
+
     @Autowired
-    public DatasourceMetadata(JdbcTemplate jdbcTemplate) {
+    public DatasourceMetadata(JdbcTemplate jdbcTemplate, DataSource dataSource) {
         this.jdbcTemplate = jdbcTemplate;
+        this.dataSource = dataSource;
```

Comment:
You don't really need this. As `JdbcTemplate` wraps `DataSource` at line 73 you can use:

`jdbcTemplate.getDataSource()`

Reply:
Done! Also, IntelliJ suggested me to wrap that datasource object with Objects.requireNonNull() method.